syntax = "proto3";

option go_package = "github.com/lncapital/torq-core/torq_cln_plugin";

// Only edit this file in the cln-plugin directory as make generate-from-proto will copy it from there

service TorqCLNPlugin {
  rpc TestConnection(EmptyMessage) returns (TestConnectionResponse) {}

  rpc InterceptChannelOpen(EmptyMessage) returns (stream InterceptChannelOpenRequest) {}

  rpc RespondChannelOpen(InterceptChannelOpenResponse) returns (EmptyMessage) {}
}

message EmptyMessage {}

message TestConnectionResponse {
    string pubkey = 1;
    string version = 2;
}

message InterceptChannelOpenRequest {
    string message_id = 1;
    double timestamp = 2;
    string peer_pubkey = 3;
    uint64 funding_amount_sat = 4;
    uint64 push_amount_msat = 5;
    uint64 dust_limit_msat = 6;
    uint64 max_htlc_value_in_flight_msat = 7;
    uint64 channel_reserve_sat = 8;
    uint64 feerate_per_kw = 9;
    uint32 to_self_delay = 10;
    uint32 max_accepted_htlcs = 11;
}

message InterceptChannelOpenResponse {
    string message_id = 1;

    double orig_timestamp = 2;

    bool allow = 3;

    optional string reject_reason = 4;
}
